crates/miden-tx/src/executor/mod.rs

```diff
@@ -254,18 +254,33 @@ where
         input_notes: InputNotes<InputNote>,
         tx_args: TransactionArgs,
     ) -> Result<TransactionInputs, TransactionExecutorError> {
-        let mut ref_blocks = validate_input_notes(&input_notes, block_ref)?;
+        let (asset_vault_keys, mut ref_blocks) = validate_input_notes(&input_notes, block_ref)?;
         ref_blocks.insert(block_ref);
 
-        let (account, block_header, blockchain) = self
+        let (account, block_header, blockchain, asset_witnesses) = self
             .data_store
-            .get_transaction_inputs(account_id, ref_blocks)
+            .get_transaction_inputs(account_id, asset_vault_keys, ref_blocks)
             .await
             .map_err(TransactionExecutorError::FetchTransactionInputsFailed)?;
 
+        // Sanity check that the data store returned an asset witness for the fee asset.
+        // This is not strictly necessary, but without the asset witness, transaction execution
+        // would fail with an obscure error.
+        let fee_asset_vault_key =
+            AssetVaultKey::from_account_id(block_header.fee_parameters().native_asset_id())
+                .expect("fee asset should be a fungible asset");
+
+        let contains_fee_asset_witness = asset_witnesses
+            .iter()
+            .any(|witness| witness.authenticates_asset_vault_key(fee_asset_vault_key));
+        if !contains_fee_asset_witness {
+            return Err(TransactionExecutorError::FetchFeeAssetWitnessFailed(fee_asset_vault_key));
+        }
+
```

Comment:
Instead of having the user honor `get_transaction_inputs`'s doc comment about having to add the fee asset witness by themselves, could we not add it to `asset_vault_keys` and have it enforced through the function's parameters?